twister: two minor fixes/enhancements

nashif opened the pull request:
- twister: handler: update missing status of failed tests
- twister: harness: match exact charachters when we detect tests

nashif committed:
twister: handler: update missing status of failed tests

Some tests report no status even without timeout, so address this case
as well.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: harness: match exact charachters when we detect tests

Avoid detecting garbage.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: fix exception message

'Handling of handler...' is a bit weird to read.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

golowanow commented on scripts/pylib/twister/twisterlib/handlers.py:
```diff
@@ -535,6 +535,7 @@ def _update_instance_info(self, harness_state, handler_time, flash_error):
             self.instance.status = harness_state
             if harness_state == "failed":
                 self.instance.reason = "Failed"
+            self.instance.add_missing_case_status("blocked", "Failed")
```
should the missing cases have reason set to the harness state instead of always 'Failed' ?
```suggestion
            self.instance.reason = "Harness state: " + harness_state
            self.instance.add_missing_case_status("blocked")
```

nashif committed:
tests: twister: use assert_called_with for multiple calls

Do not use assert_called_once_with, we can some functions more than once
and test should not prevent that.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

hakehuang reviewed (dismissed):
aside the test cases for twister need update, others looks good to me

golowanow commented on scripts/pylib/twister/twisterlib/handlers.py:
```diff
@@ -536,6 +536,7 @@ def _update_instance_info(self, harness_state, handler_time, flash_error):
             self.instance.status = harness_state
             if harness_state == "failed":
                 self.instance.reason = "Failed"
+            self.instance.add_missing_case_status("blocked", harness_state)
```
```suggestion
            self.instance.add_missing_case_status("blocked", harness_state)
```
`if` to set reason is not needed now with  add_missing_case_status() call, isn't it ?
